Source/Core/Core/IOS/USB/Bluetooth/BTEmu.cpp

```diff
@@ -412,48 +413,56 @@ bool BluetoothEmu::SendEventInquiryComplete()
+  // Additional scans will connect each remote in the proper order.
+  constexpr u8 responses_per_result = 1;
+
+  static_assert(sizeof(SHCIEventInquiryResult) - 2 +
+                    (responses_per_result * sizeof(hci_inquiry_response)) <
                 256);
 
+  const auto iter = std::find_if(m_wiimotes.begin(), m_wiimotes.end(),
+                                 std::mem_fn(&WiimoteDevice::IsInquiryScanEnabled));
+  if (iter == m_wiimotes.end())
+  {
+    // No remotes are discoverable.
+    SendEventInquiryComplete(0);
+    return false;
+  }
+
+  const auto& wiimote = *iter;
+
   SQueuedEvent event(static_cast<u32>(sizeof(SHCIEventInquiryResult) +
-                                      m_wiimotes.size() * sizeof(hci_inquiry_response)),
+                                      responses_per_result * sizeof(hci_inquiry_response)),
                      0);
 
   SHCIEventInquiryResult* inquiry_result = (SHCIEventInquiryResult*)event.buffer;
-
   inquiry_result->EventType = HCI_EVENT_INQUIRY_RESULT;
-  inquiry_result->num_responses = 0;
+  inquiry_result->num_responses = 1;
 
-  for (size_t i = 0; i != m_wiimotes.size(); ++i)
-  {
-    if (!m_wiimotes[i]->IsInquiryScanEnabled())
-      continue;
-
-    ++inquiry_result->num_responses;
-
-    u8* buffer = event.buffer + sizeof(SHCIEventInquiryResult) + i * sizeof(hci_inquiry_response);
-    hci_inquiry_response* response = (hci_inquiry_response*)buffer;
+  u8* buffer = event.buffer + sizeof(SHCIEventInquiryResult);
+  hci_inquiry_response* response = (hci_inquiry_response*)buffer;
 
-    response->bdaddr = m_wiimotes[i]->GetBD();
-    std::copy_n(m_wiimotes[i]->GetClass().begin(), HCI_CLASS_SIZE, response->uclass);
+  response->bdaddr = wiimote->GetBD();
+  response->page_scan_rep_mode = 1;
+  response->page_scan_period_mode = 0;
+  response->page_scan_mode = 0;
+  std::copy_n(wiimote->GetClass().begin(), HCI_CLASS_SIZE, response->uclass);
+  response->clock_offset = 0x3818;
 
-    response->page_scan_rep_mode = 1;
-    response->page_scan_period_mode = 0;
-    response->page_scan_mode = 0;
-    response->clock_offset = 0x3818;
-
-    DEBUG_LOG(IOS_WIIMOTE, "Event: Send Fake Inquiry of one controller");
-    DEBUG_LOG(IOS_WIIMOTE, "  bd: %02x:%02x:%02x:%02x:%02x:%02x", response->bdaddr[0],
-              response->bdaddr[1], response->bdaddr[2], response->bdaddr[3], response->bdaddr[4],
-              response->bdaddr[5]);
-  }
+  DEBUG_LOG(IOS_WIIMOTE, "Event: Send Fake Inquiry of one controller");
+  DEBUG_LOG(IOS_WIIMOTE, "  bd: %02x:%02x:%02x:%02x:%02x:%02x", response->bdaddr[0],
+            response->bdaddr[1], response->bdaddr[2], response->bdaddr[3], response->bdaddr[4],
+            response->bdaddr[5]);
 
   inquiry_result->PayloadLength =
       u8(sizeof(SHCIEventInquiryResult) - 2 +
          (inquiry_result->num_responses * sizeof(hci_inquiry_response)));
 
   AddEventToQueue(event);
-
+  SendEventInquiryComplete(1);
```

Comment:
`responses_per_result` or `inquiry_result->num_responses` instead of the hardcoded 1?